compiler/rustc_parse/src/parser/item.rs

```diff
@@ -794,6 +794,43 @@ impl<'a> Parser<'a> {
         ))
     }
 
+    /// Checks that the location of the where clause on a type alias is correct
+    /// and suggests moving if not.
+    fn check_ty_alias_where(
+        &self,
+        after_predicates: &[WherePredicate],
+        before_where_clause_present: bool,
+        before_where_clause_span: Span,
+        after_where_clause_span: Span,
+    ) {
+        if !after_predicates.is_empty() {
```

Comment:
The error should be reported if `has_where_token` for the "after" where clause is true, otherwise the code like `type A = u8 where;` will be accepted.